xds/internal/test/xds_server_integration_test.go

```diff
@@ -594,3 +598,136 @@ func (s) TestServerSideXDS_RouteConfiguration(t *testing.T) {
+					Action: v3rbacpb.RBAC_ALLOW,
+					Policies: map[string]*v3rbacpb.Policy{
+						"certain-path": {
+							Permissions: []*v3rbacpb.Permission{
+								{Rule: &v3rbacpb.Permission_UrlPath{UrlPath: &v3matcherpb.PathMatcher{Rule: &v3matcherpb.PathMatcher_Path{Path: &v3matcherpb.StringMatcher{MatchPattern: &v3matcherpb.StringMatcher_Exact{Exact: "/grpc.testing.TestService/UnaryCall"}}}}}},
+							},
+							Principals: []*v3rbacpb.Principal{
+								{Identifier: &v3rbacpb.Principal_Any{Any: true}},
+							},
+						},
+					},
+				},
+			},
+			allowEmptyCall: false,
+			allowUnaryCall: true,
+		},
+		// This test tests that the RBAC HTTP Filter hard codes the :method
+		// header to POST. Since the RBAC Configuration says to deny every RPC
+		// with a method :POST, every RPC tried should be denied.
+		{
+			name: "deny-post",
+			rbacCfg: &rpb.RBAC{
+				Rules: &v3rbacpb.RBAC{
+					Action: v3rbacpb.RBAC_DENY,
+					Policies: map[string]*v3rbacpb.Policy{
+						"post-method": {
+							Permissions: []*v3rbacpb.Permission{
+								{Rule: &v3rbacpb.Permission_Header{Header: &v3routepb.HeaderMatcher{Name: ":method", HeaderMatchSpecifier: &v3routepb.HeaderMatcher_ExactMatch{ExactMatch: "POST"}}}},
+							},
+							Principals: []*v3rbacpb.Principal{
+								{Identifier: &v3rbacpb.Principal_Any{Any: true}},
+							},
+						},
+					},
+				},
+			},
+			allowEmptyCall: false,
+			allowUnaryCall: false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			func() {
+				managementServer, nodeID, bootstrapContents, resolver, cleanup1 := setupManagementServer(t)
+				defer cleanup1()
+
+				lis, cleanup2 := setupGRPCServer(t, bootstrapContents)
+				defer cleanup2()
+
+				host, port, err := hostPortFromListener(lis)
+				if err != nil {
+					t.Fatalf("failed to retrieve host and port of server: %v", err)
+				}
+				const serviceName = "my-service-fallback"
+				resources := e2e.DefaultClientResources(e2e.ResourceParams{
+					DialTarget: serviceName,
+					NodeID:     nodeID,
+					Host:       host,
+					Port:       port,
+					SecLevel:   e2e.SecurityLevelNone,
+				})
+				inboundLis := e2e.ServerListenerWithRBACHTTPFilters(host, port, []*rpb.RBAC{test.rbacCfg})
+				resources.Listeners = append(resources.Listeners, inboundLis)
+
+				ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+				defer cancel()
+				// Setup the management server with client and server-side resources.
+				if err := managementServer.Update(ctx, resources); err != nil {
+					t.Fatal(err)
+				}
+
+				cc, err := grpc.DialContext(ctx, fmt.Sprintf("xds:///%s", serviceName), grpc.WithInsecure(), grpc.WithResolvers(resolver))
+				if err != nil {
+					t.Fatalf("failed to dial local test server: %v", err)
+				}
+				defer cc.Close()
+
+				client := testpb.NewTestServiceClient(cc)
+
+				if _, err := client.EmptyCall(ctx, &testpb.Empty{}, grpc.WaitForReady(true)); (err == nil) != test.allowEmptyCall {
```

Comment:
Check the status code `status.Code(err)`

Reply:
Switched.